torch/testing/_internal/common_methods_invocations.py

```diff
@@ -17646,33 +17639,45 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                lambda a, b, **kwargs: torch.svd_lowrank(a @ b.mT, **kwargs),
                *args, **kwargs
            ),
-           dtypes=floating_types(),
+           dtypes=floating_and_complex_types(),
            # Runs very slowly on slow gradcheck - alternatively reduce input sizes
            gradcheck_fast_mode=True,
            supports_out=False,
+           # Due to the use of randomness
            check_batched_grad=False,
            check_batched_gradgrad=False,
            check_batched_forward_grad=False,
            supports_fwgrad_bwgrad=True,
            supports_forward_ad=True,
            sample_inputs_func=sample_inputs_svd_lowrank,
            decorators=[skipCUDAIfNoCusolver, skipCPUIfNoLapack, with_tf32_off,
-                       DecorateInfo(toleranceOverride({torch.float32: tol(atol=1e-03, rtol=1e-03)}),
-                                    'TestCommon', 'test_noncontiguous_samples',
-                                    device_type='cuda')],
+                       DecorateInfo(toleranceOverride({torch.float32: tol(atol=1e-03, rtol=1e-03),
+                                                       torch.complex64: tol(atol=1e-03, rtol=1e-03)}),
+                                    'TestCommon', 'test_noncontiguous_samples'),
```

Comment:
```suggestion
                                    'TestCommon', 'test_noncontiguous_samples', device_type='cuda'),
```
? Or is the tolerance override now required for cpu device as well?

Reply:
it was also failing on CPU

Reply:
Presumably only `complex64` is failing on CPU?

Reply:
updated just for complex64. Let's hope CI passes, I had to bump tolerances on pca_lowrank on float32 for no apparent reason.

Reply:
rolling back to the previous state, it also needs a bump on CPU now.